libpod/lock/in_memory_locks.go

```diff
@@ -89,3 +89,12 @@ func (m *InMemoryManager) RetrieveLock(id uint32) (Locker, error) {
 
 	return m.locks[id], nil
 }
+
+// FreeAllLocks frees all locks
+func (m *InMemoryManager) FreeAllLocks() error {
+	for _, lock := range m.locks {
+		lock.allocated = false
+	}
+
+	return nil
+}
```

Comment:
This just feels so incredibly dangerous...  Is there anyway to wait for all locks to be freed and while waiting ensure others aren't taken out?

Reply:
We're holding the alive lock at this point, so no other processes are starting or stopping. We also have the SHM lock, so nobody else is allocating/removing at that moment.

Long-lived podman processes will retain the old lock, which means they won't break, but also probably have a different lock than newer processes. In my testing, that doesn't seem to cause problems, but I certainly don't recommend it.

In short, I think we need to mention on the man page to try and avoid using this at the same time as other Podman processes are alive, but we have protection against this doing anything nasty - the worst case is that we break any Podman processes that were running before renumber does, while launching new Podman commands works as expected.

Reply:
OK, that makes me feel better, but there are no checks in this function that validates that the alive lock and SHM lock are held.  Is there anyway we can check that fact as a safety precaution?  I'm more concerned about future development and someone not getting the locks correctly before calling this.

Reply:
It probably needs the same extended disclaimer I gave to the configuration writer functions - don't ever use this unless you have an extremely good reason and have taken the appropriate precautions